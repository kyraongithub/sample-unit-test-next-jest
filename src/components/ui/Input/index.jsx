import React from 'react';

const Input = (props) => {
  const { value, onChange, placeholder, type, label } = props;
  return (
    <>
      {label && <label htmlFor={label}>{label}</label>}
      {label && <br />}
      {type === 'textarea' ? (
        <textarea value={value} onChange={onChange} placeholder={placeholder} />
      ) : (
        <input
          id={label}
          type={type}
          value={value}
          onChange={onChange}
          placeholder={placeholder}
        />
      )}
    </>
  );
};

export default Input;
