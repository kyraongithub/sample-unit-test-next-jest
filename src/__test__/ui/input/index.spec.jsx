import Input from '@/components/ui/Input';
import { render } from '@testing-library/react';

describe('Input Component', () => {
  it('should render Input not textarea', () => {
    const { container } = render(
      <Input value="" onChange={() => {}} placeholder="test" type="password" />
    );
    expect(container).toMatchSnapshot();
  });

  it('should render Input textarea', () => {
    const { container } = render(
      <Input value="" onChange={() => {}} placeholder="test" type="textarea" />
    );
    expect(container).toMatchSnapshot();
  });
});
