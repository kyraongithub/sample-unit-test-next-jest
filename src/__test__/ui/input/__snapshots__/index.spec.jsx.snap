// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component should render Input not textarea 1`] = `
<div>
  <input
    placeholder="test"
    type="password"
    value=""
  />
</div>
`;

exports[`Input Component should render Input textarea 1`] = `
<div>
  <textarea
    placeholder="test"
  />
</div>
`;
